src/library_dylink.js

```diff
@@ -1150,30 +1150,40 @@ var LibraryDylink = {
 #endif
   },
 
-  // Async version of dlopen.
-  _emscripten_dlopen_js__deps: ['$dlopenInternal', '$callUserCallback', '$dlSetError'],
-  _emscripten_dlopen_js: (handle, onsuccess, onerror, user_data) => {
-    /** @param {Object=} e */
+  // Utility function to encapsulate promise handling with callbacks
+  $promiseToCallback: function(promise, successFnPtr, errorFnPtr, user_data) {
     function errorCallback(e) {
-      var filename = UTF8ToString(handle + {{{ C_STRUCTS.dso.name }}});
-      dlSetError(`'Could not load dynamic lib: ${filename}\n${e}`);
-      {{{ runtimeKeepalivePop() }}}
```

Comment:
It looks like the `{{{ }}}` are still removed from this. Were you able to run the tests locally? They should be failing because of this problem.

Reply:
@tlively necessary changes were missed in the previous commit. I have now pushed the latest changes. I have gone through the documentation but still doubtful on what commands and test suites needs to run to confirm for the api restructuring changes.

Reply:
Looking at CI, it looks one the failing tests is `test_pthread_dlopen_many` in the `core0` suite. To run just that test locally, you would run `test/runner core0.test_pthread_dlopen_many`. To run all the tests in `core0` (which means the core tests with optimization level O0), you would run `test/runner core0`. You can also use the `--verbose` and `--save-dir` flags to get more information and to save the files so you can manually run and debug the test yourself.